ApiDocs.Validation/MethodDefinition.cs

```diff
@@ -165,14 +165,21 @@ public void AddTestParams(string rawContent)
         /// <param name="credentials"></param>
         /// <param name="documents"></param>
         /// <returns></returns>
-        public async Task<ValidationResult<HttpRequest>> GenerateMethodRequestAsync(ScenarioDefinition scenario, string baseUrl, AuthenicationCredentials credentials, DocSet documents)
+        public async Task<ValidationResult<HttpRequest>> GenerateMethodRequestAsync(ScenarioDefinition scenario, string baseUrl, AuthenicationCredentials credentials, DocSet documents, IServiceAccount account)
```

Comment:
Since AuthenticationCredentials comes from IServiceAccount, it'd be better to simplify the calling pattern to just pass in the IServiceAccount here and use account.CreateCredentials() to get the credentials object inside this method. That way there can't be a mismatch between the account and the credentials object.